src/DatabaseManager.php

```diff
@@ -89,53 +80,14 @@ public function setDefaultConnection(string $connection)
 
     /**
      * Create the tenant database connection.
-     *
-     * @param string $databaseName
-     * @param string $connectionName
-     * @return void
-     */
-    public function createTenantConnection($databaseName, $connectionName)
-    {
-        // Create the database connection.
-        $based_on = $this->getBaseConnection($connectionName);
-        $this->app['config']["database.connections.$connectionName"] = $this->app['config']['database.connections.' . $based_on];
-
-        // Change database name.
-        $databaseName = $this->getDriver($connectionName) === 'sqlite' ? database_path($databaseName) : $databaseName;
-        $separateBy = $this->separateBy($connectionName);
-
-        $this->app['config']["database.connections.$connectionName.$separateBy"] = $databaseName;
-    }
-
-    /**
-     * Get the name of the connection that $connectionName should be based on.
-     *
-     * @param string $connectionName
-     * @return string
      */
-    public function getBaseConnection(string $connectionName): string
+    public function createTenantConnection(Tenant $tenant, $connectionName)
     {
-        return ($connectionName !== 'tenant' ? $connectionName : null) // 'tenant' is not a specific connection, it's the default
-            ?? $this->app['config']['tenancy.database.based_on']
-            ?? $this->originalDefaultConnectionName; // tenancy.database.based_on === null => use the default connection
-    }
-
-    /**
-     * Get the driver of a database connection.
-     *
-     * @param string $connectionName
-     * @return string|null
-     */
-    public function getDriver(string $connectionName): ?string
-    {
-        return $this->app['config']["database.connections.$connectionName.driver"];
```

Comment:
This class was the absolute biggest mess of the package and I'm glad we refactored a lot of this logic into the `DatabaseConfig` class. Still doesn't feel perfect to me, but a lot better than what we had before in my opinion.